ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -289,8 +284,7 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		// If in K8sJobMode, further refine the terminal condition by checking if the submitter Job has finished.
 		// See https://github.com/ray-project/kuberay/pull/1919 for reasons.
 		if rayJobInstance.Spec.SubmissionMode == rayv1.K8sJobMode {
```

Comment:
Also include `SidecarMode`. Maybe we can add an util function like `func HasSubmitter(rayjob)`.